src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -114,6 +117,28 @@ public Task StartAsync(Uri url, IChannelConnection<Message> application)
             }
         }
 
+        private IEnumerable<Message> ReadMessages(ReadOnlySpan<byte> payload)
+        {
+            if (payload.Length == 0)
+            {
+                yield break;
+            }
+
+            var messageFormat = MessageFormatter.GetFormat(payload[0]);
+            payload = payload.Slice(1);
+
```

Comment:
We could throw, but it's technically valid since the format doesn't actually require that a payload contain any frames. If we get a single `T` as a response, this code just treats that as a payload with no frames in it.